File: johansundell-go/day4/day4_test.go
package main

import "testing"

func Test_Ex1(t *testing.T) {
	input := "aaaaa-bbb-z-y-x-123[abxyz]"
	r := newRoom(input)
	if !r.isValid() {
		t.Error(input)
	}
}

func Test_Ex2(t *testing.T) {
	input := "a-b-c-d-e-f-g-h-987[abcde]"
	r := newRoom(input)
	if !r.isValid() {
		t.Error(input)
	}
}

func Test_Ex3(t *testing.T) {
	input := "not-a-real-room-404[oarel]"
	r := newRoom(input)
	if !r.isValid() {
		t.Error(input)
	}
}

func Test_Ex4(t *testing.T) {
	input := "totally-real-room-200[decoy]"
	r := newRoom(input)
	if r.isValid() {
		t.Error(input)
	}
}

func Test_Ex5(t *testing.T) {
	r := room{}
	r.id = 343
	r.name = "qzmt-zixmtkozy-ivhz"
	if r.decryptName() != "very encrypted name" {
		t.Error()
	}
}

File: falkdahl-python/Day4-test.py
import unittest, Day4

class TestDay4(unittest.TestCase):

    def testRealRooms(self):
        self.assertTrue(Day4.isRealRoom('aaaaa-bbb-z-y-x-123[abxyz]'))
        self.assertTrue(Day4.isRealRoom('a-b-c-d-e-f-g-h-987[abcde]'))
        self.assertTrue(Day4.isRealRoom('not-a-real-room-404[oarel]'))

    def testFakeRooms(self):
        self.assertFalse(Day4.isRealRoom('totally-real-room-200[decoy]'))

if __name__ == '__main__':
    unittest.main()
